File: Design_Pattern_and_Principles/ProxyPatternExample/ProxyTest.java
public class ProxyTest {
    public static void main(String[] args) {
        Image image = new ProxyImage("photo_render_4k.png");

        System.out.println("First call: Image needs to be loaded.");
        image.display();

        System.out.println("\nSecond call: Image should load instantly via proxy cache.");
        image.display();
    }
}

File: Design_Pattern_and_Principles/ProxyPatternExample/ProxyTest.cs
using System;

public class ProxyTest {
    public static void Main(string[] args) {
        IImage image = new ProxyImage("photo_render_4k.png");

        Console.WriteLine("First call: Image needs to be loaded.");
        image.Display();

        Console.WriteLine("\nSecond call: Image should load instantly via proxy cache.");
        image.Display();
    }
}